pymatgen/io/vasp/sets.py

```diff
@@ -526,20 +526,19 @@ def incar(self) -> Incar:
                 if key.startswith("LDAU"):
                     del incar[key]
 
-        # Modify LMAXMIX if you have d or f electrons present.
-        # Note that if the user explicitly sets LMAXMIX in settings it will
-        # override this logic.
-        # Previously, this was only set if Hubbard U was enabled as per the
-        # VASP manual but following an investigation it was determined that
-        # this would lead to a significant difference between SCF -> NonSCF
-        # even without Hubbard U enabled. Thanks to Andrew Rosen for
-        # investigating and reporting.
+        # Modify LMAXMIX if you have d or f electrons present. Note that if the user
+        # explicitly sets LMAXMIX in settings it will override this logic.
+        # Previously, this was only set if Hubbard U was enabled as per the VASP manual
+        # but following an investigation it was determined that this would lead to a
+        # significant difference between SCF -> NonSCF even without Hubbard U enabled.
+        # Thanks to Andrew Rosen for investigating and reporting.
         if "LMAXMIX" not in settings:
+            blocks = {site.specie.block for site in structure}
             # contains f-electrons
-            if any(el.Z > 56 for el in structure.composition):
+            if "d" in blocks:
                 incar["LMAXMIX"] = 6
             # contains d-electrons
-            elif any(el.Z > 20 for el in structure.composition):
```

Comment:
Should be `if "d" in blocks`. I made a mistake in the WIP Atomate2 PR (fixed now).